mood_bot.py

```diff
@@ -0,0 +1,726 @@
+    time_stamp = mood_diary_retrieval(author, 'date', 'last_entry')
+    await context.send(f"You have an unfinished mood check from {time_stamp}.")
+
+
+def mood_stamp_maker(author):
+    mood_stamp = mood_diary_retrieval(author, 'type', 'last_entry')
+    return mood_stamp
+
+
+def rating_stamp_maker(author):
+    rating_stamp = mood_diary_retrieval(author, 'rating', 'last_entry')
+    return rating_stamp
+
+
+async def mood_rating_message(context, mood_stamp):
+    await context.send(f'You left off at rating. The mood type was {mood_stamp}')
+
+
+async def mood_why_message(context, mood_stamp, rating_stamp):
+    await context.send(f"You left off at the explanation. The mood type was {mood_stamp} "
+                       f"and the rating was {rating_stamp}.")
+
+
+@mood_bot.command(name='check', help='Run a mood check.')
+async def mood_check(context):
+    """This function starts a mood check by recording current time and date and asking the user how they feel."""
+    global date_gate, type_gate
+    author = context.author
+    if date_gate == 'Open':
+        entry_time = datetime.datetime.now()
+        mood_diary_storage(author, 'date', entry_time.strftime('%Y/%m/%d %H:%M:%S'))
+        await context.send("How are you feeling, " + str(author) + "?\n" +
+                           "Please select from the following mood types:\n" +
+                           str(mood_choices) + "\n" +
+                           "Please respond by typing 'mood_type' followed by your selected mood.")
+        date_gate = 'Closed'
+        type_gate = 'Open'
+    else:
+        """If the date_gate isn't open, it checks to see which gate is open.
+        This means that another mood check was in progress and is unfinished."""
+        await time_stamp_message(context, author)
+        if type_gate == 'Open':
+            await context.send(mood_type_message)
+        else:
+            mood_stamp = mood_stamp_maker(author)
+            if rating_gate == 'Open':
+                await mood_rating_message(context, mood_stamp)
+            elif why_gate == 'Open':
+                rating_stamp = rating_stamp_maker(author)
+                await mood_why_message(context, mood_stamp, rating_stamp)
+
+
+async def mood_type_error(context, mood):
+    await context.send(mood_type_error_message)
+    raise me.InputMoodError(mood_type_error_message, mood)
+
+
+@mood_bot.command(name='type')
+async def mood_type_selection(context, mood_type):
+    """Allows the user to answer the mood_check function with their mood selection.
+    And asks the user to rate their mood from 1 to 10."""
+    global type_gate, rating_gate
+    author = context.author
+    if type_gate == 'Open':
+        if mood_type.lower() not in mood_choices:
+            await mood_type_error(context, mood_type.lower())
+        else:
+            mood_diary_storage(author, 'type', mood_type)
+            await context.send(f"On a scale from 1 to 10, how {mood_type} do you feel?\n" +

```

Comment:
we can move all these strings to constants above in all these messages, would be nice to take them out of this part of the app

Reply:
I have converted all strings to constants and have stored them in another file.